tests/PHPStanTests/UnsafeStrictGroupsCallRuleTest.php

```diff
@@ -44,6 +44,10 @@ public function testRule(): void
                 'The isMatchStrictGroups call is potentially unsafe as $matches\' type could not be inferred.',
                 86,
             ],
+            [
+                'The isMatchAllStrictGroups call is unsafe as match groups "test", "1" are optional and may be null.',
```

Comment:
I am not 100% sure this error is expected, as I don't got the use-case of the rule

Reply:
This new warning is correct/fine.

Reply:
The point is, the method returns only non-null matches, but if one is null it throws, so it isn't safe to use with nullable match groups, which is what this warns about. In the test though you want nullable match groups to ensure they appear not-null in $matches.